File: netflix-prize-graph-example/python/ratings_by_viewer.py
import sys
from netflix_dal import get_ratings_by_viewer

if len(sys.argv) != 2:
    print('Usage: ratings_by_viewer <viewer_id>')
    exit(1)

viewer_id = sys.argv[1]
try:
    result = get_ratings_by_viewer(viewer_id)

    if len(result) == 0:
        print(f'The viewer {viewer_id} does not have any ratings in the database.')
        exit(0)

    for record in result:
        rating = record.get('rating')
        movie = record.get('movie')
        date_rated = rating.get('dateRated').iso_format() # neo4j.time.Date object-to-string conversion.
        print(f"{date_rated}: “{movie.get('title')}” got a {rating.get('rating')}.")
except ValueError:
    print(f'Sorry, something went wrong. Please ensure that “{viewer_id}” is a valid viewer ID.')

File: netflix-prize-graph-example/nodejs/ratings-by-viewer.mjs
import { getRatingsByViewer } from './netflix-dal.mjs'

if (process.argv.length !== 3) {
  console.log('Usage: ratings-by-viewer <viewerId>')
  process.exit(1)
}

const viewerId = process.argv[2]
try {
  const result = await getRatingsByViewer(viewerId)

  if (result.length === 0) {
    console.log(`The viewer ${viewerId} does not have any ratings in the database.`)
    process.exit(0)
  }

  result.forEach(record => {
    const rating = record.get('rating').properties
    const movie = record.get('movie').properties

    // There’s more than meets the eye here: `console.log(record, rating, movie)` to see.
    console.log(`${rating.dateRated}: “${movie.title}” got a ${rating.rating}.`)
  })
} catch (error) {
  console.error(`Sorry, something went wrong. Please ensure that “${viewerId}” is a valid viewer ID.`)
}

// Explicitly exit sync since we are asynchronous at this point.
process.exit(0)
